src/server.py

```diff
@@ -1 +1,39 @@
 """Server.py for python echo server."""
+
+import socket
+import sys
+
+
+def server():
+    """Place server into listening mode wating for connection."""
+    server = socket.socket(socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_TCP)
+    address = ('127.0.0.1', 5030)
+    server.bind(address)
+    server.listen(5)
+    print("Waiting for connection...")
+    while True:
+        try:
+            conn, addr = server.accept()
+            if conn:
+                return_message = ''
+                buffer_length = 8
+                message_complete = False
+                while not message_complete:
+                    part = conn.recv(buffer_length)
+                    return_message += part.decode('utf8')
```

Comment:
So you're decoding each chunk as you get it. Since unicode characters can take up multiple bytes, what happens when a multiple-byte unicode character isn't received all the way in one chunk? i.e. the buffer length cuts it in half?

Rather than decoding one chunk at a time, what would be a better approach?

And anyway, for the purpose of this echo server, do you really need to decode the message at the server end, only to re-encode it and send it back?